jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/JdbcRecordIterator.java

```diff
@@ -52,7 +51,8 @@ public class JdbcRecordIterator implements Iterator<Map<String, Object>> {
   private String[] hiveColumnNames;
   List<TypeInfo> hiveColumnTypesList;
 
-  public JdbcRecordIterator(Connection conn, PreparedStatement ps, ResultSet rs, Configuration conf) throws HiveJdbcDatabaseAccessException {
+  public JdbcRecordIterator(GenericJdbcDatabaseAccessor accessor, Connection conn,
```

Comment:
General comment, not something this patch introduced. I wish these Connection/PreparedStatement/ResultSet objects were not being passed in but instead were accessible thru the accessor object. Makes a bit more sense IMHO. This code can just call into the accessor object for cleanup as well instead of doing it here.

What do you think? (I am not suggesting that we do this in this patch)

Reply:
We get the Connection/PreparedStatement and release them on demand in accessor, it would be not easy to manage the lifecycle of them in accessor.
I'm trying to make DatabaseAccessor extend AutoCloseable, so that we can release the `DataSource` when necessary.